addons/io/org.openhab.io.azureiothub/src/main/java/org/openhab/io/internal/azureiothub/CloudClient.java

```diff
@@ -0,0 +1,169 @@
+ *
+ * This class provides communication between openHAB and Azure IoT Hub.
+ * implementation is taken from https://docs.microsoft.com/en-us/azure/iot-hub/iot-hub-java-java-getstarted
+ *
+ * @author Niko Tanghe - Initial contribution
+ * @author Kai Kreuzer - Code cleanup
+ *
+ */
+
+public class CloudClient {
+
+    private final Logger logger = LoggerFactory.getLogger(CloudClient.class);
+
+    private String connectionstring;
+    private boolean commandEnabled;
+    private EventPublisher eventPublisher;
+
+    private AzureDevices azureDeviceStore;
+    private EventHubClient azureClient;
+    private Object lockobj = new Object();
+
+    /**
+     * Constructor of CloudClient
+     *
+     * @param connectionstring the connectionstring to the Azure IoT Hub
+     * @param eventPublisher
+     * @throws IOException
+     *
+     */
+    public CloudClient(String connectionstring, boolean commandEnabled, EventPublisher eventPublisher)
+            throws Exception {
+        this.connectionstring = connectionstring;
+        this.commandEnabled = commandEnabled;
+        this.eventPublisher = eventPublisher;
+
+        azureDeviceStore = new AzureDevices(this.connectionstring);
+
+        if (this.commandEnabled) {
+            azureClient = receiveMessages("0");
+        }
+    }
+
+    public void sendItemUpdate(String deviceId, String state) {
+
+        AzureDevice device;
+        try {
+            device = azureDeviceStore.getDevice(deviceId);
+        } catch (Exception e) {
+            logger.error("Failed to obtain azure device", e);
+            return;
+        }
+
+        setItemState(device, state);
+    }
+
+    private void setItemState(AzureDevice azureDevice, String state) {
+        AzureEventCallback callback = new AzureEventCallback();
+
+        AzureDatapoint datapoint = new AzureDatapoint();
+
+        Device device = azureDevice.getDevice();
+        if (device == null) {
+            logger.error("Invalid device connection for device, can not send item state update");
+            return;
+        }
+
+        datapoint.deviceId = device.getDeviceId();
+        datapoint.value = state;
+
+        Message msg = new Message(datapoint.serialize());
+
+        azureDevice.sendMessage(msg, callback, lockobj);
+
+        synchronized (lockobj) {
+            try {
+                lockobj.wait();
+            } catch (InterruptedException e) {
+                // done
+            }
+        }
+    }
+
+    public void shutdown() {
+        if (azureClient != null) {
+            try {
+                azureClient.closeSync();
+            } catch (ServiceBusException e) {
+                logger.error("failed to close IoT client", e);
+            }
+        }
+    }
+
+    private EventHubClient receiveMessages(final String partitionId) {
+        EventHubClient client = null;
+        try {
+            client = EventHubClient.createFromConnectionStringSync(this.connectionstring);
+        } catch (Exception e) {
```

Comment:
Do you really need to catch the Exception-class or are there more specific exceptions you could catch like the IOException or maybe only RuntimeException's (which is not that good either). The current code and also catching RuntimeException's might 'hide' programming errors like NullPointerExceptions occurring.

https://www.google.nl/search?q=java%20why%20not%20to%20catch%20exception